packages/connection/src/class-manager.php

```diff
@@ -2678,4 +2683,26 @@ public function get_signed_token( $token ) {
 		return join( ' ', $header_pieces );
 	}
 
+	/**
+	 * If connection is active, add the list of plugins using connection to the heartbeat (except Jetpack itself)
+	 *
+	 * @param array $stats The Heartbeat stats array.
+	 * @return array $stats
+	 */
+	public function add_stats_to_heartbeat( $stats ) {
+
+		if ( ! $this->is_active() ) {
+			return $stats;
+		}
+
+		$active_plugins_using_connection = Plugin_Storage::get_all();
+		foreach ( array_keys( $active_plugins_using_connection ) as $plugin_slug ) {
+			if ( 'jetpack' !== $plugin_slug ) {
+				$stats_group           = isset( $active_plugins_using_connection['jetpack'] ) ? 'combined-connection' : 'standalone-connection';
+				$stats[ $stats_group ] = $plugin_slug;
```

Comment:
I think we want to add the `$plugin_slug` to the array here instead of setting the value.

```suggestion
				$stats[ $stats_group ][] = $plugin_slug;
```

Do you think we should try to add unit tests for this method?

Reply:
hi @kbrown9, thank you so much for your thorough review (as always).

I will look into this tomorrow morning, incorporate your suggestion and think of a way to test it.